src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputColumnLimit.java

```diff
@@ -50,4 +51,54 @@ int test1(int badFormat1, int badFormat2, final int badFormat3) throws java.lang
 
   // Very long url with valid href: href    = "www.google.com/search?hl=en&q=java+style+guide+checkstyle+check+href+length+limit&btnG=Google+Search"
   int validHrefWithWhiteSpaces = 54;
+
+  @SuppressWarnings({"LineLength"})
+  int aaaarealllllllllllllllllyyyyyyyyyyylllllllloooooooooooooooonnnnnnnnnnnnnnnnnggggggggggvvvvvvaarriaaabllee = 99; // ok, suppression used, check name in camelcase
+
+  @SuppressWarnings({"linelength"})
+  int aaaarealllllllllllllllllyyyyyyyyyyylllllllloooooooooooooooonnnnnnnnnnnnnnnnnggggggggggvvvvvvaarriaaabllee2 = 99; // ok, suppression used, check name in lowercase
+
+  @SuppressWarnings({"linelength"})
+  void longggggggggggggggggggggmethoooooooooooooooooooooodddddddddddddddddddddddddddddddddddddd(int x, int y) {} // ok
+
+  @SuppressWarnings({"linelength"})
+  void testingParametersNames(int areallllllyyyyyyyyyyyyyyyyyylonnnnggggggggggname, int anotherlongnameblahblahblah) {} // ok
+
+  @SuppressWarnings({"linelength"})
+  InputColumnLimit inputColumnLimit = (InputColumnLimit) ((Object) new com.google.checkstyle.test.chapter4formatting.rule44columnlimit.InputColumnLimit()); // ok
+
+  @SuppressWarnings({"linelength"})
+  void testing() {
+    try {
+      throwExceptionBasedOnCondition();
+
+    } catch (IOException | NullPointerException | ArrayIndexOutOfBoundsException | ClassNotFoundException ex) { // ok, got suppressed
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @SuppressWarnings({"linelength"})
+  void throwExceptionBasedOnCondition()
+      throws IOException, NullPointerException, ArrayIndexOutOfBoundsException, ClassNotFoundException { // ok, got suppressed
+    // Above is violation of the rule, user should not use suppression for such cases
+    int condition = new java.util.Random().nextInt(3);
+    switch (condition) {
+      case 0:
+        throw new IOException("Test IOException");
+      case 1:
+        throw new NullPointerException("Test NullPointerException");
+      case 2:
+        throw new ArrayIndexOutOfBoundsException("Test ArrayIndexOutOfBoundsException");
+      case 3:
+        throw new ClassNotFoundException("Test ClassNotFoundException");
+      default:
+    }
+  }
+
+  @SuppressWarnings({"linelength"})
+  class LonggggggggggggggggggggggggggggClassssssssssssssssssssssssssNameeeeeeeeeeeeeeSoooooBoooooorrriinngggg {
+    void longggggggggggggggggggggmethoooooooooooooooooooooodddddddddddddddddddddddddddddddddddddd(int x, int y) {} // ok
+
+    int aaaarealllllllllllllllllyyyyyyyyyyylllllllloooooooooooooooonnnnnnnnnnnnnnnnnggggggggggvvvvvvaarriaaabllee = 99; // ok
+  }
```

Comment:
adding suppression to any block of code such as classes, methods, enums, etc will disable the specified Check for that whole block.

Reply:
please restore few examples of `@SuppressWarnings` user will decide himself what to use, we just offer multiple approaches.